src/overlays/actors/ovl_En_Hg/z_en_hg.c

```diff
@@ -310,76 +333,87 @@ void func_80BCF95C(EnHg* this, PlayState* play) {
         } else {
             if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
                 switch (this->animIndex) {
-                    case 3:
-                        this->animIndex = 4;
-                        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, 4);
+                    case HG_ANIM_LEAN_FORWARD:
+                        this->animIndex = HG_ANIM_REACH_FORWARD;
+                        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, HG_ANIM_REACH_FORWARD);
                         break;
-                    case 5:
-                        this->animIndex = 6;
-                        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, 6);
+
+                    case HG_ANIM_CURL_UP:
+                        this->animIndex = HG_ANIM_CROUCHED_PANIC;
+                        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, HG_ANIM_CROUCHED_PANIC);
                         break;
                 }
             }
         }
+
         switch (this->animIndex) {
-            case 3:
-            case 4:
+            case HG_ANIM_LEAN_FORWARD:
+            case HG_ANIM_REACH_FORWARD:
                 func_800B9010(&this->actor, NA_SE_EN_HALF_REDEAD_LOOP - SFX_FLAG);
                 break;
-            case 5:
-            case 6:
+
+            case HG_ANIM_CURL_UP:
+            case HG_ANIM_CROUCHED_PANIC:
                 func_800B9010(&this->actor, NA_SE_EN_HALF_REDEAD_SCREAME - SFX_FLAG);
                 break;
-            case 7:
-                if ((this->unk218 == 0) || (this->unk218 == 2)) {
+
+            case HG_ANIM_PANIC:
+                if ((this->cutsceneIndex == HG_CS_FIRST_ENCOUNTER) ||
+                    (this->cutsceneIndex == HG_CS_ALTERNATIVE_ENCOUNTER)) {
                     func_800B9010(&this->actor, NA_SE_EN_HALF_REDEAD_SCREAME - SFX_FLAG);
                 }
                 break;
         }
+
         Cutscene_ActorTranslateAndYaw(&this->actor, play, actionIndex);
         return;
+
     } else if (play->csCtx.state == 0) {
-        func_80BCF354(this);
+        EnHg_SetupIdle(this);
     }
+
     this->cutscenes[3] = 0x63;
 }
 
-void func_80BCFC0C(EnHg* this, PlayState* play) {
+void EnHg_WaitForPlayerAction(EnHg* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
 
+    // If the player has gone upstairs this interaction will not trigger
     if (this->actor.colChkInfo.health == 1 && !(fabsf(this->actor.playerHeightRel) >= 80.0f)) {
```

Comment:
The fabsf condition looks like it might be an early return, could you do 
```suggestion
    if ((this->actor.colChkInfo.health != 1) || (fabsf(this->actor.playerHeightRel) >= 80.0f)) {
    return;
    }
```
?